core/src/main/java/com/linecorp/armeria/common/logging/TextLogFormatter.java

```diff
@@ -188,6 +192,34 @@ public String formatRequest(RequestOnlyLog log) {
         }
     }
 
+    private static void maybeAppendConnectionTimings(@Nullable ClientConnectionTimings timings,
+                                                     StringBuilder buf) {
+        if (timings == null) {
+            return;
+        }
+        buf.append(", {ClientConnection startTime=");
+        TextFormatter.appendEpochMicros(buf, timings.connectionAcquisitionStartTimeMicros());
+        if (timings.dnsResolutionDurationNanos() >= 0) {
+            buf.append(", dnsResolutionStartTime=");
+            TextFormatter.appendEpochMicros(buf, timings.dnsResolutionStartTimeMicros());
```

Comment:
What do you think of using durations instead?
Most users are primarily concerned with durations and would have to do additional calculations otherwise.

Reply:
What do you also think of using shorter names to make the full log more concise?
e.g. `{Connections: acquisition=xxx, dns=xxx, socket=xxx, tls=xxx, pendingAcquisition=xxx}`

Reply:
> What do you think of using durations instead?
Most users are primarily concerned with durations and would have to do additional calculations otherwise.

Interesting, I thought most users would be more concerned with the timings than the durations.
I also thought that it is easier to deduce durations if timings information is available. (I'm not so sure calculating timings based on durations is easier)

Having said this, I guess I'm fine either way. Let me know which way you prefer more.

> What do you also think of using shorter names to make the full log more concise?

Sure, sounds good

Reply:
I'm not also so strong on this. @ikhoon Do you have a preference?

Reply:
jfyi, I'm actually now thinking that date information isn't very important - connection information is likely not long-living so the timestamps is more important.

What do you think of the format `epochMillis(duration)`?

e.g.
```
[creqId=52f3109f, chanId=15b501e1, laddr=0:0:0:0:0:0:0:1%lo0:41702, raddr=0:0:0:0:0:0:0:1%lo0:80][h2c://127.0.0.1/#GET] Request: {startTime=2024-06-05T06:18:00.311Z(1717568280311000), {ClientConnection total=1717568280309000(15042ns), dns=1717568280309000(5417ns), pending=1717568280309000(1250ns), socket=1717568280309000(2042ns), tls=1717568280309000(1709ns)}, length=0B, duration=153µs(153458ns), scheme=none+h2c, name=GET, headers=[:method=GET, :path=/]}
```

Reply:
> I thought most users would be more concerned with the timings than the durations.
(I'm not so sure calculating timings based on durations is easier)

There's a total duration composed of TLS, socket, pending and DNS durations. By examining these durations, Users can easily infer which stage takes longer. Do you have any specific scenarios in mind where users might have difficulty identifying the problem with this information?

Reply:
> What do you think of the format epochMillis[duration]?

The log is a bit longer, but I believe it's acceptable. 👍

Reply:
> Do you have any specific scenarios in mind where users might have difficulty identifying the problem with this information?

Although each tag is understood by use, I thought it might be difficult for users to know which action occurs in what order and when.

Having said this, I guess we can change the format later if needed. Updated!

Reply:
What do you think of putting both time and duration for completeness for clarity, at least for JSON? I understand it is worth keeping the log concise, but for JSON output, I think we should provide all the information for machines.

Reply:
Sounds good. 👍 
I'm also okay with including both time and duration for JSON and text formats, in case a user complains about any discrepancies.

Reply:
Updated to the following formats:

##### TEXT
Format
```
epochMicros[durationNanos]
```
Sample
```
[creqId=995494c0, chanId=b84b93af, laddr=0:0:0:0:0:0:0:1%lo0:36505, raddr=0:0:0:0:0:0:0:1%lo0:80][h2c://127.0.0.1/#GET] Request: {startTime=2024-06-07T06:53:00.092Z(1717743180092000), Connection: {total=1717743180085000[5291ns], dns=1717743180085000[208ns], pending=1717743180085000[1084ns], socket=1717743180085000[1625ns], tls=1717743180085000[208ns]}, length=0B, duration=104µs(104125ns), scheme=none+h2c, name=GET, headers=[:method=GET, :path=/]}
```

##### JSON
Format
```
Connection: {
  "tls": {
      "durationNanos": 123
      "startMicros": 123
  }
}
```
Sample
```
{
  "type": "request",
  "startTime": "2024-06-07T06:51:39.156Z(1717743099156000)",
  "connection": {
    "total": {
      "durationNanos": 5625,
      "startTimeMicros": 1717743099142000
    },
    "dns": {
      "durationNanos": 166,
      "startTimeMicros": 1717743099142000
    },
    "pending": {
      "durationNanos": 1291,
      "startTimeMicros": 1717743099142000
    },
    "socket": {
      "durationNanos": 1667,
      "startTimeMicros": 1717743099142000
    },
    "tls": {
      "durationNanos": 209,
      "startTimeMicros": 1717743099142000
    }
  },
  "length": "0B",
  "duration": "88333ns",
  "scheme": "none+h2c",
  "name": "GET",
  "headers": {
    ":method": "GET",
    ":path": "/"
  }
}
```

Reply:
Optional) How about using ISO 8601 instead of Unix time in the format? I don't want to waste too much of your time here, but unix time seems unreadable for text format.
```
Connection: {total=2023-10-01T12:19:40.085000Z[5291ns], dns=2023-10-01T12:19:40.085000Z[208ns], pending=2023-10-01T12:19:40.085000Z[1084ns], socket=2023-10-01T12:19:40.085000Z[1625ns], tls=2023-10-01T12:19:40.085000Z[208ns]}
```

Reply:
If we do that, we should remove `Micros` from the property names.

Reply:
> How about using ISO 8601 instead of Unix time in the format

Updated `TextLogFormatter` to return the format in ISO 8601

```
[creqId=09867aa3, chanId=91d541ac, laddr=0:0:0:0:0:0:0:1%lo0:54796, raddr=0:0:0:0:0:0:0:1%lo0:80][h2c://127.0.0.1/#GET] Request: {startTime=2024-06-10T02:49:11.015Z(1717987751015000), Connection: {total=2024-06-10T02:49:11.007000Z[5292ns], dns=2024-06-10T02:49:11.007000Z[167ns], pending=2024-06-10T02:49:11.007000Z[1083ns], socket=2024-06-10T02:49:11.007000Z[1625ns], tls=2024-06-10T02:49:11.007000Z[208ns]}, length=0B, duration=89916ns, scheme=none+h2c, name=GET, headers=[:method=GET, :path=/]}
```

> If we do that, we should remove Micros from the property names.

I assume this is pertinent to `JsonLogFormatter`. I imagine users may want to perform some tasks (e.g. calculate distributions) with JSON data, so I think it's probably easier if we just include the time unit in the name and just set the number for `JsonLogFormatter`. Let me know if this doesn't make sense though

> Is microsecond-granularity data format common? I've never seen such a case, so I think users might have a little bit of difficulty while parsing it? So.. I lean towards using microseconds number.

I've just tried to use the highest resolution possible given that `ClientConnectionTimings` probably completes in short succession. Let me know if you prefer to use `millisecond` timestamps instead of ISO 8601 with microsecond precision though.

Reply:
> One concern is: Is microsecond-granularity data format common?

Since Java 9, `Instant.toString()` converts the time to ISO 8601 with microseconds precision. 
```
System.out.println(Instant.now());
// 2024-06-10T12:12:58.834503Z
```